### PR TITLE
net-misc/ena: new package

### PR DESCRIPTION
**Pull request by sormy**
Here is an ebuild for `ena` kernel module that is needed to enable network on new Amazon Instances.
See more here: <https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/enhanced-networking-ena.html>

I verified that amd64 version could be compiled, loaded as kernel module and actually adds support for network on new Amazon EC2 instances like c5.2xlarge. I verified that this driver is also could be compiled on x86 platform and could be loaded as kernel driver with no isssue, but did not test on actual instance.

Package-Manager: Portage-2.3.40, Repoman-2.3.9
RepoMan-Options: --force

### CODE REVIEW

File: net-misc/ena/ena-1.5.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit linux-info linux-mod eutils
+
+DESCRIPTION="Amazon EC2 Elastic Network Adapter (ENA) kernel driver"
+HOMEPAGE="https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/enhanced-networking-ena.html"
+SRC_URI="https://github.com/amzn/amzn-drivers/archive/ena_linux_${PV}.zip -> ${P}-linux.zip"
```
[the remark this answers is not in the document]
Comment (sormy) [reply net-misc/ena/ena-1.5.3.ebuild]: I could keep it as it. But please note, "fine" is non-formal definition and should be avoided in formal requirements unless you want all new contributors to go away to other distros or never submit PR to main repo.

File: net-misc/ena/ena-1.5.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+LICENSE="GPL-2"
```
[the remark this answers is not in the document]
Comment (sormy) [reply net-misc/ena/ena-1.5.3.ebuild]: Not all files in repo are dual licensed.

File: net-misc/ena/ena-1.5.3.ebuild [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+SLOT="0"
+KEYWORDS="amd64 x86"
```
[the remark this answers is not in the document]
Comment (sormy) [reply net-misc/ena/ena-1.5.3.ebuild]: This is default kernel module maintained by Amazon and included into their Amazon Linux distribution. Network won't work on EC2 C5-like instances without this module. This package has no dependencies and not used as dependency in any other package. It won't hurt any profile and it is stable.

File: net-misc/ena/ena-1.5.3.ebuild [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+S="${WORKDIR}/amzn-drivers-ena_linux_${PV}/kernel/linux/ena"
+
+MODULE_NAMES="ena(net:${S}:${S})"
+BUILD_TARGETS="all"
+
+pkg_setup() {
+	linux-mod_pkg_setup
+	BUILD_PARAMS="CONFIG_MODULE_SIG=n KERNEL_DIR=${KV_DIR}"
```
Comment (sormy) [inline net-misc/ena/ena-1.5.3.ebuild]: There is an issue here `CONFIG_MODULE_SIG=n`. I will remove it.

File: net-misc/ena/metadata.xml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+    <maintainer type="person">
+        <email>art.sormy@gmail.com</email>
+        <name>Artem Butusov</name>
+    </maintainer>
```
[the remark this answers is not in the document]
Comment (sormy) [reply net-misc/ena/metadata.xml]: done

File: net-misc/ena/ena-1.5.3.ebuild (hunk 1 above)
[the remark this answers is not in the document]
Comment (mgorny) [reply net-misc/ena/ena-1.5.3.ebuild]: Are we supposed to write 1000-page book explaining why people shouldn't do a lot of different unnecessary things? I'm sorry but my imagination is limited and I can't predict all the things people could do for no reason.

File: net-misc/ena/ena-1.5.3.ebuild (hunk 2 above)
[the remark this answers is not in the document]
Comment (mgorny) [reply net-misc/ena/ena-1.5.3.ebuild]: Next time, it would be nice to support that with evidence rather than expecting me to verify your claims.

**Comment by mgorny** [comment]
Now please squash the commits into one with final commit messages I could put into the repo.

File: net-misc/ena/ena-1.5.3.ebuild (hunk 2 above)
[the remark this answers is not in the document]
Comment (sormy) [reply net-misc/ena/ena-1.5.3.ebuild]: What kind of evidence is needed? There is file in the root of kernel module sources with explicitly stated GPL-2: https://github.com/amzn/amzn-drivers/blob/master/kernel/linux/ena/COPYING . By the way, there is no requirement to provide evidence. If you expect evidence to be provided then update "how to submit PR" wiki page.

File: net-misc/ena/ena-1.5.3.ebuild (hunk 1 above)
[the remark this answers is not in the document]
Comment (sormy) [reply net-misc/ena/ena-1.5.3.ebuild]: If 1000-page book is required to make all requirements formal then there is a problem with requirements. By the way, yes, all formal requirements should be documented somewhere. Ideally repoman could provide a link with explanation why requirement is here. Without explanation all these statements look like personal preferences and... I guess I don't need to explain why decisions based on personal preferences won't help to grow the community?!

**Comment by sormy** [comment]
Nice, there is no squash requirement in PR docs. ok.

---

A few words for Gentoo portage's elected leader: All current contributors will eventually gone. Family, lack of interest, no time... The work you do is amazing but you need more contributors. Informal-requirements won't help community to grow to support the project. This is most-likely my first and last contribution to gentoo portage based on experience I got, as an example. Think wise and have a good good luck.

**Comment by sormy** [comment]
Just to confirm. Do you want me to just squash two commits and resign them or also pull HEAD, rebase on top of it, squash two commits into one and resign?

**Comment by mgorny** [comment]
Please fix your attitude if you'd like to contribute. Your tone makes it unpleasant to work with you.

**Comment by sormy** [comment]
This is unprofessional way to answer on professional concerns. 

My tone is professional. I'm sure I'm asking right questions. By the way I did not get any reasonable answers on them. It seems like pointing to the problems is not considered as appropriate question and causing frustration. But... this is a path to nowhere. Ignoring the problems won't solve them. 

Making contributors praying on the knees to get PR approved, or asking for apologize for asking considered as inappropriate questions will eventually kill the community. I have no way to change that, but you have.

By the way, I still would like to wish a good luck and growth to gentoo community, including personally you, Mike. Who knows, probably you just have a hard time in real life and my PR came just not in the right time. ;-)

**Pull request closed without merging**